### ISSUE TITLE
OTA Update not working - stackoverflow!

### ISSUE THREAD

**Issue by melvinpmathew** [espressif/arduino-esp32#1359]
the OTA updater was working a week back, now it seems to be failing. Any idea?

I had recently made some changes to have keypair on NVS(Non volatile storage) , did it affect anything?

### Hardware:
Board:							ESP32 Wrover

### Description:
Content-Type: application/octet-stream
Content-Length: 641776
Server: AmazonS3
Connection: close

Got application/octet-stream payload.
Got 641776 bytes from server
contentLength : 641776, isValidContentType : 1

[D][Updater.cpp:123] begin(): OTA Partition: app0
Begin OTA. This may take 2 - 5 mins to complete. Things might be quite for a while.. Patience!
Got application/octet-stream payload.
Got 641776 bytes from server
contentLength : 641776, isValidContentType : 1
Begin OTA. This may take 2 - 5 mins to complete. Things might be quite for a while.. Patience!
Guru Meditation Error of type InstrFetchProhibited occurred on core  1. Exception was unhandled.
Register dump:
PC      : 0x004e4f61  PS      : 0x00060f30  A0      : 0x800d28a8  A1      : 0x3ffcff10  
A2      : 0x3ffc3324  A3      : 0x3ffda324  A4      : 0x00001000  A5      : 0x00001000  
A6      : 0x00ff0000  A7      : 0xff000000  A8      : 0x800d287c  A9      : 0x3ffcfef0  
A10     : 0x3ffc3324  A11     : 0x3ffdb324  A12     : 0x00000000  A13     : 0x0000003f  
A14     : 0x00000000  A15     : 0x00000000  SAR     : 0x00000018  EXCCAUSE: 0x00000014  
EXCVADDR: 0x004e4f60  LBEG    : 0x4000c2e0  LEND    : 0x4000c2f6  LCOUNT  : 0xffffffff  

Backtrace: 0x004e4f61:0x3ffcff10 0x400d28a5:0x3ffcff30 0x400d2396:0x3ffcff50 0x400d163e:0x3ffcff70 0x400d17f1:0x3ffcffc0 0x40118e53:0x3ffd0000

PC: 0x004e4f61
EXCVADDR: 0x004e4f60

Decoding stack results
0x400d28a5: Stream::readBytes(char*, unsigned int) (C:\Program Files at x86)\Arduino\hardware\espressif\esp32\cores\esp32\Stream.cpp line 239
0x400d2396: UpdateClass::writeStream(Stream&) (C:\Program Files at x86)\Arduino\hardware\espressif\esp32\cores\esp32/Stream.h line 103
0x400d163e: execOTA() at C:\Users\e222985\Documents\Arduino\AWS_S3_OTA_Update1111/AWS_S3_OTA_Update1111.ino line 158
0x400d17f1: setup() at C:\Users\e222985\Documents\Arduino\AWS_S3_OTA_Update1111/AWS_S3_OTA_Update1111.ino line 213
0x40118e53: loopTask(void*) (C:\Program Files at x86)\Arduino\hardware\espressif\esp32\cores\esp32\main.cpp line 15

![image](https://user-images.githubusercontent.com/24640290/39314203-58847f14-4939-11e8-8cbe-5c6430856a88.png)

```

**Comment by melvinpmathew** [comment]
Any help guys!

**Comment by everslick** [comment]
@melvinpmathew you did not provide any code. From the information you gave, it looks like a bug in your sketch. For general help, pls use this [gitter channel](https://gitter.im/espressif/arduino-esp32).

If you do not think this is a bug in the ESP32 Arduino Core, pls close this issue. Thanks!

**Comment by melvinpmathew** [comment]
Sorry guys for the delay in response. Had a newborn addition to the family this week and didn't expect the sleep cycle change :).

This issues was resolved, I updated the xtensa tool chain and ESP32 core drivers, and after this the OTA updates are working normally again! My clients are real happy to get back the cloud updates.

@stickbreaker : When I had this issue, I did try releasing the NVS and other global memory object and still it was failing at the same point.

@everslick : I did not provide the code since this issue was replicable on the example S3 OTA code. Once I updated the core and toolchain the example worked again and thus my sketch. Thanks. I will close this now.
